aggregator_core/src/task.rs

```diff
@@ -39,11 +39,20 @@ pub enum QueryType {
     /// Time-interval: used to support a collection style based on fixed time intervals.
     TimeInterval,
 
-    /// Fixed-size: used to support collection of batches as quickly as possible, without aligning
-    /// to a fixed batch window.
+    /// Fixed-size: used to support collection of batches as quickly as possible, without the
+    /// latency of waiting for batch time intervals to pass, and with direct control over the number
+    /// of reports per batch.
     FixedSize {
         /// The maximum number of reports in a batch to allow it to be collected.
         max_batch_size: u64,
+        /// If present, reports will be separated into different batches by timestamp, such that
+        /// the client timestamp interval duration will not exceed this value. The minimum and
+        /// maximum allowed report timestamps for each batch will be multiples of this value as
+        /// well. This must be a multiple of the task's time precision.
+        ///
+        /// This is an implementation-specific configuration parameter, and not part of the query
+        /// type as defined in DAP.
+        batch_time_window_size: Option<Duration>,
```

Comment:
Something to note is that taskprov has no way of encoding this.
```rust
struct {
    QueryType query_type;         /* I-D.draft-ietf-ppm-dap-04 */
    [...]
    select (QueryConfig.query_type) {
        case time_interval: Empty;
        case fixed_size:    uint32 max_batch_size;
    }
} QueryConfig;
```
--we don't have space to specify `batch_time_window_size`. I think it's not a real problem given our use cases for taskprov, just making a note.

Reply:
Yep, I think this is OK given how we want to deploy taskprov.